rs/execution_environment/tests/dts.rs

```diff
@@ -3063,3 +3063,105 @@ fn dts_abort_after_dropping_memory_on_state_switch() {
     let result = env.execute_ingress(canister_id, "update", vec![]).unwrap();
     assert_eq!(result, WasmResult::Reply(vec![]));
 }
+
+const WRITE_MORE_THAN_1G_WAT: &str = r#"
+(module
+    (import "ic0" "msg_reply" (func $msg_reply))
+    (func (export "canister_update write")
+        (local $i i32)
+        (local.set $i (i32.const 1073745920)) ;; 1GiB + 4096
+        (loop $loop
+            (i32.store (local.get $i) (i32.const 1))
+            (br_if $loop (local.tee $i (i32.sub (local.get $i) (i32.const 4096))))
+        )
+        (call $msg_reply)
+    )
+    (memory 16385) ;; 1GiB + 65536
+)"#;
```

Comment:
Would it be possible to use the [WatCanisterBuilder](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/test_utilities/execution_environment/src/wat_canister.rs?L291-293) instead of hardcoding the wat file? Since we have this utility we might as well use it for better readability.

Reply:
It appears `WatCanisterBuilder` doesn't currently support loops or declaring 1GiB of memory. I'll chat with Maksym on Thursday, and merge the PR as is for now...

Reply:
Sounds good.